src/main/diff.mli

```diff
@@ -0,0 +1,86 @@
+module Field : sig
```

Comment:
**Suggestion:** Some module-level docs wouldn't hurt here -- it'd probably make it a little more clear what this library intends to do. A README would be good too, but at a minimum perhaps some module-level docs would do in a pinch!

Reply:
Added some comments and a README 👍